Blog

EricChavarria opened the pull request:
I made the Blogs page. I realized while making this that we will need an individual blog page as well... The Blog itself is another component for this page called BlogItem! The only static thing on this page at the end of the day will be the Musical musings title at the top. the blogs will have different photos and snippets based on each different blog
<img width="336" alt="Screen Shot 2020-02-16 at 12 55 04 PM" src="https://user-images.githubusercontent.com/27781451/74609815-b12f6080-50bb-11ea-8535-8bc764253f47.png">
<img width="372" alt="Screen Shot 2020-02-16 at 12 55 10 PM" src="https://user-images.githubusercontent.com/27781451/74609816-b7bdd800-50bb-11ea-9191-ebebda7fae03.png">

Commit:
Made the contact page

Commit:
Make conditional class

Commit:
Create Services Page

Commit:
get rid of id's

Haon1919 committed:
Merge branch 'develop' into contact

Commit:
Use the FormInput Component

Haon1919 committed:
Calendar html + render helper

Begin html and add a render function to construct the calendar with proper dates for the respective month.

joverton2697 committed:
Landing Page

Landing page is mostly finished, final images and text need to be decided. Header has blank space underneath

joverton2697 committed:
Implementing feedback

Commit:
Make changes to Contact Page

Commit:
Make styling change to services

Commit:
Change content

Commit:
Get started on Blog Page

Commit:
Finish up Blog Component

Commit:
get Started on Forgot Password Page

Commit:
Finish Forgot Password Page

Haon1919 commented on src/components/BlogItem.tsx:
```diff
@@ -0,0 +1,16 @@
+import { Link, useHistory } from 'react-router-dom';
+import styles from '../styles/BlogItem.module.scss'
+
+const BlogItem : React.FC= () => {
```
You are going to want to pass a blog item as a prop. Create a type in a type file for blogs as I have done with other type instances.

Haon1919 commented on src/pages/Blog.tsx:
```diff
@@ -1,9 +1,19 @@
+import musicalMusings from '../assets/images/musical-musings.jpeg';
+import styles from '../styles/Blog.module.scss';
 
 export const Blog : React.FC= () => {
```
Instantiate an array of blog items and map over them then within the map pass the blog item to the blog item component. This will make life easier for use when we connect it to the backend, all we will have to do is set a state variable. Let me know if you have any questions about this.

Haon1919 commented on src/components/BlogItem.tsx:
```diff
@@ -0,0 +1,16 @@
+const BlogItem : React.FC= () => {
+    return (
+        <section className={styles.blog_container}>
+            <img src={catImage} alt="Cat" className={styles.blog_image}/>
```
The source of the image may come back with the network request. We should ask seth about this.

Haon1919 commented on src/components/BlogItem.tsx:
```diff
@@ -0,0 +1,16 @@
+    return (
+        <section className={styles.blog_container}>
+            <img src={catImage} alt="Cat" className={styles.blog_image}/>
+            <p className={styles.blog_snippet}>Amagna urna viverra ipsum, vitae egesnte non elementum gravida, lectus tortor ultrices dolor, eu imperdiet ipsum odio efficitur felis. Quisque sit amet bibendum leo, vitae auctor diam. In id cursus ante. Cras vel semper lacus.</p>
```
Text will be apart of the blog item that is passed in as a prop.

Haon1919 commented on src/components/BlogItem.tsx:
```diff
@@ -0,0 +1,16 @@
+        <section className={styles.blog_container}>
+            <img src={catImage} alt="Cat" className={styles.blog_image}/>
+            <p className={styles.blog_snippet}>Amagna urna viverra ipsum, vitae egesnte non elementum gravida, lectus tortor ultrices dolor, eu imperdiet ipsum odio efficitur felis. Quisque sit amet bibendum leo, vitae auctor diam. In id cursus ante. Cras vel semper lacus.</p>
+            <Link to="/blog" className={styles.blog_link}>View</Link>
```
The link will be a little strange. We will probably send it to a blog page followed by the id of the post. like /blog/{id}

Haon1919 committed:
Got the repository battle ready

Haon1919 committed:
Got the repository battle ready

Finish calendar, add schedule lesson and lesson page, fix header z-index issue, add fake login functionality.

Haon1919 committed:
Merge remote-tracking branch 'origin/Calendar' into Calendar

Haon1919 committed:
Merge remote-tracking branch 'origin/blog' into PresentationBranch

Haon1919 committed:
Merge branch 'forgot-password' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Services' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/contact' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Landing' into PresentationBranch

Haon1919 committed:
Fix Login link in header

Fix login link in header by prefacing it with a slash

Commit:
Fix typo and mini bug

Haon1919 committed:
General tab remains collapsed when user logs out

Fix bug where if the user logs out and the Students tab is open, the general tab remains collapsed.

Haon1919 committed:
Merge remote-tracking branch 'origin/PresentationBranch' into PresentationBranch

Haon1919 committed:
Nonrequired form blocking submit if fails validation

If a non required field fails validation, make sure that when they remove the text it removes the failing field from the list of failing fields.

Haon1919 committed:
Minor form failing error persists for non minors

Fix error where when the birth date of the user made them a minor,and the user entered an invalid email for the guardian email then switched the birth date to make them an adult and filled in all of the other forms correctly, the  invalid guardian email still disallowed them from submitting even though they guardian email is not an option for adults.

Haon1919 committed:
Merge branch 'PresentationBranch' into blog
